### PR TITLE
Set up build infrastructure to bring in lightbulb-supporting types

### PR DESCRIPTION
**Pull request by lgolding**
Also:
- Update to latest xunit.
- Adjust some unit tests to quiet VSTRHD analyzer warnings.

### CODE REVIEW

File: src/Sarif.Viewer.VisualStudio/FixSuggestedAction.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Language.Intellisense;
+
+namespace Microsoft.Sarif.Viewer
+{
+    class FixSuggestedAction : ISuggestedAction
```
Comment (GabeDeBacker) [inline src/Sarif.Viewer.VisualStudio/FixSuggestedAction.cs:13]: >ISuggestedAction [](start = 31, length = 16)

Does this actually derive from IDisposable, or are we adding it?

File: src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSource.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.Sarif.Viewer
+{
+    internal class FixSuggestedActionsSource : ISuggestedActionsSource
```
Comment (GabeDeBacker) [inline src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSource.cs:14]: >ISuggestedActionsSource [](start = 47, length = 23)

Does this actually derive from IDisposable. I'm assuming the fixes will be disposable due to persistent spans being disposable

File: src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSourceProvider.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.Sarif.Viewer
+{
+    [Export(typeof(ISuggestedActionsSourceProvider))]
+    [Name(FixActionCategoryName)]
+    [ContentType("text")]
+    internal class FixSuggestedActionsSourceProvider : ISuggestedActionsSourceProvider
+    {
+        internal const string FixActionCategoryName = "SARIF fix suggestion";
```
Comment (GabeDeBacker) [inline src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSourceProvider.cs:18]: > "SARIF fix suggestion" [](start = 53, length = 23)

Put in RESX please

Comment (GabeDeBacker) [inline src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSourceProvider.cs:18]: >internal [](start = 8, length = 8)

Do we actually need access to this? Can this be private?

File: src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSourceProvider.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+        [Import(typeof(ITextStructureNavigatorSelectorService))]
+        internal ITextStructureNavigatorSelectorService NavigatorService { get; set; }
```
Comment (GabeDeBacker) [inline src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSourceProvider.cs:21]: >internal [](start = 8, length = 8)

Do we actually need access to this? Can this be private?

File: src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSource.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        public FixSuggestedActionsSource(FixSuggestedActionsSourceProvider fixSuggestedActionsSourceProvider, ITextView textView, ITextBuffer textBuffer)
```
Comment (GabeDeBacker) [inline src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSource.cs:16]: >FixSuggestedActionsSourceProvider fixSuggestedActionsSourceProvider, [](start = 41, length = 68)

I would personally avoid circular references and just pass this class only what it needs.

File: src/Sarif.Viewer.VisualStudio/FixSuggestedAction.cs (hunk 1 above)
Comment (GabeDeBacker) [inline src/Sarif.Viewer.VisualStudio/FixSuggestedAction.cs:13]: add internal

**Comment by GabeDeBacker** [comment]
Adding XML documentation on public methods is something we should start doing :)

/// <inheritdoc/> 

is your friend for interface implementations

File: src/Sarif.Viewer.VisualStudio/FixSuggestedAction.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        public string IconAutomationText => throw new NotImplementedException();
+
+        public string InputGestureText => throw new NotImplementedException();
+
+        public bool HasPreview => throw new NotImplementedException();
+
+        public void Dispose()
+        {
+            throw new NotImplementedException();
+        }
```
Comment (GabeDeBacker) [inline src/Sarif.Viewer.VisualStudio/FixSuggestedAction.cs:30]: Be consistent, use expression bodies for now

File: src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSource.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+#pragma warning restore 0067
+
+        public void Dispose()
+        {
+        }
+
+        public IEnumerable<SuggestedActionSet> GetSuggestedActions(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken)
+        {
+            yield return new SuggestedActionSet(new FixSuggestedAction[0]);
+        }
```
Comment (GabeDeBacker) [inline src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSource.cs:38]: for now just do "yield break;" :)

**Review by GabeDeBacker — APPROVED** [review]
:shipit:

File: src/Sarif.Viewer.VisualStudio/FixSuggestedAction.cs (hunk 1 above)
Comment (lgolding) [reply src/Sarif.Viewer.VisualStudio/FixSuggestedAction.cs:13]: `ISuggestedAction` extends `IDisposable`.

---
In reply to: [509475047](https://github.com/microsoft/sarif-visualstudio-extension/pull/228#discussion_r509475047) [](ancestors = 509475047)

File: src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSourceProvider.cs (hunk 3 above)
Comment (lgolding) [reply src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSourceProvider.cs:18]: Is there a way to use string resources in attribute arguments? Thats's where this is needed, in Line 14 above.

---
In reply to: [509476895](https://github.com/microsoft/sarif-visualstudio-extension/pull/228#discussion_r509476895) [](ancestors = 509476895)

File: src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSource.cs (hunk 2 above)
Comment (lgolding) [reply src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSource.cs:14]: Yes, it does.

---
In reply to: [509475985](https://github.com/microsoft/sarif-visualstudio-extension/pull/228#discussion_r509475985) [](ancestors = 509475985)

File: src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSource.cs (hunk 5 above)
Comment (lgolding) [reply src/Sarif.Viewer.VisualStudio/FixSuggestedActionsSource.cs:16]: I'm sympathetic to that argument, but the fact that it's `public` in the sample I copied suggests that somebody in VS needs it. I'd rather leave the argument and the public property in place for now, the remove them and wonder why it mysteriously fails. I'll leave a comment to remove it once we actually see the light bulbs.

---
In reply to: [509479052](https://github.com/microsoft/sarif-visualstudio-extension/pull/228#discussion_r509479052) [](ancestors = 509479052)

File: src/Sarif.Viewer.VisualStudio.Interop/SarifViewerInterop.cs [hunk 8]
```diff
@@ -79,6 +79,8 @@ public bool IsViewerExtensionInstalled
         {
             get
             {
+                ThreadHelper.ThrowIfNotOnUIThread();
```
Comment (jeffersonking) [inline src/Sarif.Viewer.VisualStudio.Interop/SarifViewerInterop.cs:82]: Curious: Did these analyzers just recently kick in?

Comment (lgolding) [reply src/Sarif.Viewer.VisualStudio.Interop/SarifViewerInterop.cs:82]: Yes. Even though we are using a 15-level SDK, I've discovered that it's implicitly pulling in 16-level analyzers.

---
In reply to: [509536067](https://github.com/microsoft/sarif-visualstudio-extension/pull/228#discussion_r509536067) [](ancestors = 509536067)

**Pull request merged**